src/couch_views/src/couch_views.erl

```diff
@@ -81,7 +83,10 @@ read_view(Db, Mrst, ViewName, Callback, Acc0, Args) ->
         after
             UpdateAfter = Args#mrargs.update == lazy,
             if UpdateAfter == false -> ok; true ->
-                couch_views_jobs:build_view_async(TxDb, Mrst)
+                % the current transaction is a read only, add the job in a new
```

Comment:
Is there ever a case where we want to create a job in the same transaction then? Seems like moving the new transaction bit inside the couch_views_jobs module would be better?

Reply:
Yes we want to add it when we adding the index job in the update transaction. If we do it in a different transaction there it causes FDB errors because couch_jobs and fabric2 are each checking the metadata_version so they clash.